kr8s/_objects.py

```diff
@@ -824,6 +826,33 @@ async def uncordon(self) -> None:
         """
         await self.async_patch({"spec": {"unschedulable": False}})
 
+    async def taint(self, key: str, value: str, *, effect: str) -> None:
+        """Taint a node."""
+        if effect.endswith("-"):
+            effect = effect[:-1]
+            await self.async_patch(
+                [
+                    {
+                        "op": "remove",
+                        "path": "/spec/taints",
+                        "value": [{"key": key, "value": value, "effect": effect}],
+                    }
+                ],
+                type="json",
+            )
+        else:
+            await self.async_patch(
+                {"spec": {"taints": [{"effect": effect, "key": key, "value": value}]}}
+            )
```

Comment:
If there are already taints will this not replace them all with this new taint? We might need to do `"op": "add"` here instead.

Reply:
Good point.

Looking at `kubectl taint nodes ... my-key=value1:NoSchedule -v10`, I see it makes two calls, one to get all the taints, and another one update it. I updated this PR to copy what `kubectl` does.